packages/proxy/test/unit/http/request-middleware.spec.ts

```diff
@@ -78,20 +79,39 @@ describe('http/request-middleware', () => {
 
       await testMiddleware([MaybeAttachCrossOriginCookies], ctx)
 
-      expect(ctx.req.headers['cookie']).to.equal('exist=ing')
+      expect(ctx.req.headers['cookie']).to.equal('request=cookie')
     })
 
-    it('is prepends cookie jar cookies to request', async () => {
+    it('prepends cookie jar cookies to request', async () => {
       const ctx = getContext()
 
       await testMiddleware([MaybeAttachCrossOriginCookies], ctx)
 
-      expect(ctx.req.headers['cookie']).to.equal('new=one; exist=ing')
+      expect(ctx.req.headers['cookie']).to.equal('jar=cookie; request=cookie')
     })
 
-    function getContext () {
+    // @see https://github.com/cypress-io/cypress/issues/22751
+    it('does not double up cookies on request if the cookie exists on the request and in the cookie jar', async () => {
+      const ctx = getContext(['jar=cookie', 'request=cookie'])
+
+      await testMiddleware([MaybeAttachCrossOriginCookies], ctx)
+
+      expect(ctx.req.headers['cookie']).to.equal('jar=cookie; request=cookie')
+    })
+
+    // The idea here being if the same key/value were to be set in the cookie jar, but the path/samesite/secure etc is different.
+    // TODO: I wonder if we need to do additional logic here when attaching cookies to requests based on which path is most specific?
```

Comment:
Is this something we actually need to worry about, ie which cookie to apply if the key/value is the same but the path is different. In the case of path, do we send the cookie with the more specific path?